### PR TITLE
Replace gfx_draw_string_clipped() calls with DrawTextEllipsised

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
I don't think we should be using the void args at all in the refactor. This should be the end of us using void args.

File: src/openrct2-ui/windows/GuestList.cpp [hunk 1]
```diff
@@ -857,7 +857,7 @@ static void window_guest_list_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi,
                     std::memcpy(
                         gCommonFormatArgs, _window_guest_list_groups_arguments[i].args,
                         std::min(sizeof(gCommonFormatArgs), sizeof(_window_guest_list_groups_arguments[i].args)));
-                    gfx_draw_string_left_clipped(dpi, format, gCommonFormatArgs, COLOUR_BLACK, { 0, y }, 414);
+                    DrawTextEllipsised(dpi, { 0, y }, 414, format, gCommonFormatArgs, COLOUR_BLACK);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/GuestList.cpp:860]: @duncanspumpkin OK, then tell me how to do this line, for example.

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/GuestList.cpp:860]:
```Formatter ft(_window_guest_list_groups_arguments[i].args);
DrawTextEllipsised(dpi, { 0, y }, 414, format, ft, COLOUR_BLACK);
```

**Comment by Gymnasiast** [comment]
I have now moved all the remaining calls to use a Formatter. This did involve changing some seemingly unrelated code as well.

**Review by duncanspumpkin — COMMENTED** [review]
I think this looks much better. Good work. I'll have a closer look next week at the changes unless someone beats me to it

**Review by duncanspumpkin — APPROVED** [review]

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 2]
```diff
@@ -784,22 +783,20 @@ static void window_loadsave_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi, i
...
 
         // Print formatted modified date, if this is a file
         if (_listItems[i].type == TYPE_FILE)
         {
             ft = Formatter::Common();
             ft.Add<rct_string_id>(STR_STRING);
             ft.Add<char*>(_listItems[i].date_formatted.c_str());
-            gfx_draw_string_right_clipped(
-                dpi, stringId, gCommonFormatArgs, COLOUR_BLACK, { dateAnchor - DATE_TIME_GAP, y }, maxDateWidth);
+            DrawTextEllipsised(dpi, { dateAnchor - DATE_TIME_GAP, y }, maxDateWidth, stringId, ft, COLOUR_BLACK);
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/LoadSave.cpp:794]: This needs text alignment set to right.

File: src/openrct2-ui/windows/GuestList.cpp [hunk 3]
```diff
@@ -811,8 +815,8 @@ static void window_guest_list_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi,
                                 if (thought->freshness > 5)
                                     break;
 
-                                peep_thought_set_format_args(&peep->Thoughts[j]);
-                                gfx_draw_string_left_clipped(dpi, format, gCommonFormatArgs, COLOUR_BLACK, { 118, y }, 329);
+                                peep_thought_set_format_args(&peep->Thoughts[j], ft);
+                                DrawTextEllipsised(dpi, { 118, y }, 329, format, ft, COLOUR_BLACK);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/GuestList.cpp:819]: The `Formatter` should have been reset here

File: src/openrct2-ui/windows/InstallTrack.cpp [hunk 4]
```diff
@@ -241,8 +241,7 @@ static void window_install_track_paint(rct_window* w, rct_drawpixelinfo* dpi)
         if (!gTrackDesignSceneryToggle)
         {
             // Scenery not available
-            gfx_draw_string_centred_clipped(
-                dpi, STR_DESIGN_INCLUDES_SCENERY_WHICH_IS_UNAVAILABLE, nullptr, COLOUR_BLACK, screenPos, 368);
+            DrawTextEllipsised(dpi, screenPos, 308, STR_BLACK_STRING, Formatter::Common(), COLOUR_BLACK, TextAlignment::CENTRE);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/InstallTrack.cpp:244]: Didn't we stop showing `STR_DESIGN_INCLUDES_SCENERY_WHICH_IS_UNAVAILABLE` here?

File: src/openrct2-ui/windows/Guest.cpp [hunk 5]
```diff
@@ -1160,8 +1160,8 @@ void window_guest_overview_paint(rct_window* w, rct_drawpixelinfo* dpi)
     }
 
     screenPos.x = widget->width() - w->list_information_type;
-    peep_thought_set_format_args(&peep->Thoughts[i]);
-    gfx_draw_string_left(&dpi_marquee, STR_WINDOW_COLOUR_2_STRINGID, gCommonFormatArgs, COLOUR_BLACK, { screenPos.x, 0 });
+    peep_thought_set_format_args(&peep->Thoughts[i], ft);
+    DrawTextBasic(&dpi_marquee, { screenPos.x, 0 }, STR_WINDOW_COLOUR_2_STRINGID, ft, COLOUR_BLACK);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Guest.cpp:1164]: `ft` should have been reset before this call

**Pull request merged**